node/src/components/contract_runtime/announcements.rs

```diff
@@ -78,14 +77,14 @@ impl Display for ContractRuntimeAnnouncement {
 pub(super) async fn step_success<REv>(
     effect_builder: EffectBuilder<REv>,
     era_id: EraId,
-    execution_effect: ExecutionEngineExecutionEffect,
+    execution_journal: ExecutionJournal,
 ) where
     REv: From<ContractRuntimeAnnouncement>,
 {
     effect_builder
         .schedule_regular(ContractRuntimeAnnouncement::StepSuccess {
             era_id,
-            execution_effect: (&execution_effect).into(),
+            execution_effect: (&execution_journal).into(),
```

Comment:
This gets so confusing now - there are two `ExecutionEffect` types: one in `types/src/execution_result.rs` and another in `execution_engine/src/core/engine_state/execution_effect.rs`. There exists `impl From<ExecutionJournal> for ...` for both of them, for the former one it preserves the order of transforms while for the latter it does not 😵

Reply:
Can we at least use the `From` impl here more explicitly - like `casper_types::ExecutionEffect::from(&execution_journal)`?

Reply:
Agreed, very confusing.